### PR TITLE
Better errors and docs about node-specific config keys of pools and networks

### PR DESCRIPTION
**Pull request by freeekanayaka**
Signed-off-by: Free Ekanayaka <free.ekanayaka@canonical.com>

### CODE REVIEW

File: lxd/networks.go [hunk 1]
```diff
@@ -115,7 +115,8 @@ func networksPost(d *Daemon, r *http.Request) Response {
 		// value for the storage config is 'bridge.external_interfaces'.
 		for key := range req.Config {
 			if !shared.StringInSlice(key, db.NetworkNodeConfigKeys) {
-				return SmartError(fmt.Errorf("Invalid config key '%s'", key))
+				return SmartError(
+					fmt.Errorf("Config key '%s' may not be used as node-specific key", key))
```
Comment (stgraber) [inline lxd/networks.go]: Can we avoid the wrapping here, makes grepping unnecessarily harder in this case (we don't have a max-width requirement in our codebase)?

File: lxd/storage_pools.go [hunk 2]
```diff
@@ -136,7 +136,8 @@ func storagePoolsPost(d *Daemon, r *http.Request) Response {
 	// storage config are the ones in StoragePoolNodeConfigKeys.
 	for key := range req.Config {
 		if !shared.StringInSlice(key, db.StoragePoolNodeConfigKeys) {
-			return SmartError(fmt.Errorf("Invalid config key '%s'", key))
+			return SmartError(
+				fmt.Errorf("Config key '%s' may not be used as node-specific key", key))
```
Comment (stgraber) [inline lxd/storage_pools.go]: Can we avoid the wrapping here, makes grepping unnecessarily harder in this case (we don't have a max-width requirement in our codebase)?

File: lxd/networks.go (hunk 1 above)
Comment (freeekanayaka) [reply lxd/networks.go]: Done, hope it's what you meant.

File: lxd/storage_pools.go (hunk 2 above)
Comment (freeekanayaka) [reply lxd/storage_pools.go]: Done, hope it's what you meant.

**Pull request merged**